internal/utils/conditions.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024
```

Comment:
Not super important but it would be nice if this was placed in package other than `internal/utils/` as it can easily become a mishmash of different functions.